.github/workflows/test-spyre.yml

```diff
@@ -1,29 +1,133 @@
 name: test-sypre
 
-on: pull_request
+on:
+  workflow_dispatch:
+  push:
+    paths-ignore:
+      - "**.md"
+  pull_request:
+    paths-ignore:
+      - "**.md"
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  TEST_IMAGE_NAME: "vllm-spyre"
 
 jobs:
-  test-spyre:
+  build:
     runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+    env:
+      CACHE_IMAGE: "ghcr.io/vllm-project/vllm-spyre:test-cache"
+      CACHE_REGISTRY: "ghcr.io"
```

Comment:
Do we need a repo admin to set up any secrets in order to use the ghcr registry for caching?

Reply:
GHCR does not need additional secrets to be set up. For Quay.io we would need to do that, though.